samples/image/canvas.cpp

```diff
@@ -44,27 +34,23 @@ wxBEGIN_EVENT_TABLE(MyCanvas, wxScrolledWindow)
     EVT_PAINT(MyCanvas::OnPaint)
 wxEND_EVENT_TABLE()
 
-MyCanvas::MyCanvas( wxWindow *parent, wxWindowID id,
-                    const wxPoint &pos, const wxSize &size )
-    : wxScrolledWindow( parent, id, pos, size, wxSUNKEN_BORDER )
+MyCanvas::MyCanvas(wxWindow* parent)
+    : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxSUNKEN_BORDER)
     , m_bmpSmileXpm(smile_xpm)
     , m_iconSmileXpm(smile_xpm)
 {
     my_horse_ani = nullptr;
-    m_ani_images = 0 ;
+    m_ani_images = 0;
 
-    SetBackgroundColour(* wxWHITE);
+    SetBackgroundColour(*wxWHITE);
 
-    wxBitmap bitmap( 100, 100 );
+    wxBitmap bitmap;
+    bitmap.CreateWithLogicalSize(FromDIP(wxSize(100, 100)), GetDPIScaleFactor());
```

Comment:
Wouldn't `CreateWithDIPSize()` work here?

Reply:
I reverted these changes and always create a bitmap with scale 1. This way bitmap loaded from file and the bitmap created for the canvas will always have the same scale, even after DPI events.